File: asset-management/src/main/java/com/example/asset_management/asset/dto/CreateAssetRequest.java
package com.example.asset_management.asset.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;

@Data
public class CreateAssetRequest {

    private Integer manufacturerYear;
    @NotBlank(message="Brand is required")
    private String brand;
    @NotBlank(message="Model is required")
    private String model;
    @NotBlank(message="Host Name is required")
    private String hostName;
    @NotBlank(message="Serial Number is required")
    private String serialNumber;

    private String cpu;

    private String ram;

    private String ssd;

    private String operatingSystem;

    private String officeVersion;

    private String powerAdapterSerial;

    private LocalDate warrantyStartDate;

    private LocalDate warrantyEndDate;
    @NotNull(message="Department is required")
    private Long departmentId;
    @NotNull(message="Engineer is required")
    private Long engineerId;
}

File: asset-management-ui/src/app/core/models/create-asset-request.ts
export interface CreateAssetRequest {

  manufacturerYear: number | null;

  brand: string;

  model: string;

  hostName: string;

  serialNumber: string;

  cpu: string | null;

  ram: string | null;

  ssd: string | null;

  operatingSystem: string | null;

  officeVersion: string | null;

  powerAdapterSerial: string | null;

  warrantyStartDate: string | null;

  warrantyEndDate: string | null;

  departmentId: number;

  engineerId: number;

}
